pages/learn/more/anti-patterns.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Development Anti-patterns
+---
+# How to break {{ $names.os.lower }} (development anti-patterns)
+
+As you are building your application and preparing for production, there are a number of things to keep in mind that can cause problems on a device. Many of these items may seem obvious or even desired, but often cause more trouble than they are worth.
+
+## Networking
+One of the biggest causes of issues during deployment is either a misconfigured local network or otherwise a network that does not adhere to the {{ $names.os.lower }} [networking requirements][networking-reqs].
+
+### Connectivity dependent on container (e.g. kernel module for wifi dongle, or udev rule for modem)
+Since there can be cases where application containers fail to start, having any networking configuration depend on those application containers without a fallback is generally discouraged. Using a project like [wifi-connect][wifi-connect] can help configure the networking stack after launch, but devices should be connected as reliably as possible at boot.
+
+### Iptables rules that block supervisor/VPN traffic
+It is enticing to use custom iptables rules to filter traffic before it hits the application, to limit the surface of the API. Those rules combined with host networking however can prevent the supervisor or VPN from successfully connecting to {{ $names.cloud.lower }}, and therefore should be combined with extreme scrutiny. For a full list of networking requirements, consult [this document][networking-reqs].
+
+### Running docker-in-docker with host networking and iptables enabled
+(can be folded into the previous issue, leaving for completeness at review)
+
+### Manually changing the system clock, or blocking NTP requests
+Since {{ $names.os.lower }} & the supervisor communicate with {{ $names.cloud.lower }} using an HTTP API, it is important that time is synchronized on the device to ensure that certificates are trusted. {{ $names.os.upper }} provides a [number of mechanisms][time-sync] to keep time as up-to-date as possible, but ensuring [NTP is accessible over the network][networking-reqs] is critical.
+
+## Local storage
+
+### Writing to files in the container file system (and not a volume)
+Commonly, users misunderstand the distinction between the container's file system and a volume. Named volumes are preserved across updates unless specifically dereferenced or destroyed, while anything written to the container's file system will be purged during the next application update. For an in-depth comparison of the two, it is recommended to complete the following [services masterclass][services-mc]. Additionally, the [supervisor provides an API][supervisor-api] to manage named volumes.
```

Comment:
I understand there are performance and disk space implications as well, especially in the AUFS case. Suggestion:
> [...] during the next application update. In addition, the container's filesystem often uses the [AUFS driver](https://docs.docker.com/storage/storagedriver/aufs-driver/), which typically has worse performance both in disk space and CPU utilization when compared to a named volume's standard Linux [ext4 filesystem](https://en.wikipedia.org/wiki/Ext4). For an in-depth comparison [...]

Reply:
TIL! thanks for pointing that out :)